chain/ethereum/src/network.rs

```diff
@@ -73,6 +87,42 @@ impl EthereumNetworkAdapters {
         self.adapters
             .retain(|adapter| adapter.adapter.provider() != provider);
     }
+
+    pub fn call_or_cheapest(
+        &self,
+        capabilities: Option<&NodeCapabilities>,
+    ) -> anyhow::Result<Arc<EthereumAdapter>> {
+        match self.call_only_adapter()? {
+            Some(adapter) => Ok(adapter),
+            None => self.cheapest_with(capabilities.unwrap_or(&NodeCapabilities {
+                // Archive is required for call_only
+                archive: true,
+                traces: false,
+            })),
+        }
+    }
+
+    pub fn call_only_adapter(&self) -> anyhow::Result<Option<Arc<EthereumAdapter>>> {
+        if self.call_only_adapters.is_empty() {
+            return Ok(None);
+        }
+
+        let adapters = self
+            .call_only_adapters
+            .iter()
+            .min_by_key(|x| Arc::strong_count(&x.adapter))
+            .ok_or(anyhow!("no available call only endpoints"))?;
+
+        // TODO: This will probably blow up a lot sooner than [limit] amount of
+        // subgraphs, since we probably use a few instances.
+        if Arc::strong_count(&adapters.adapter) >= adapters.limit {
```

Comment:
Yeah, I used the `Arc::strong_count` mostly as a hack; great if this works, but we might run into the limits of this approach.

Reply:
I think this is another part we probably should change properly, this method was already being used, I don't think it's very good for this part of the code because we have a lot of different copies of the underlying adapters for a single subgraph.